pkg/kube/kube_handler_test.go

```diff
@@ -1088,137 +1093,523 @@ func setClientsetData(cs fakeCombinedClientset, namespaceNames []existingNs, err
 	)
 }
 
-func TestValidateAppRepository(t *testing.T) {
+func TestGetValidator(t *testing.T) {
 	const kubeappsNamespace = "kubeapps"
-	getValidationCliAndReqTests := []struct {
-		name                       string
-		requestData                string
-		requestNamespace           string
-		expectedURLs               []string
-		expectedHeaders            http.Header
-		expectedError              error
-		expectedReqResolutionError error
+	testCases := []struct {
+		name              string
+		appRepo           *v1alpha1.AppRepository
+		expectedValidator HttpValidator
+		expectedError     error
```

Comment:
Updated this test case to focus on unit-testing `getValidator` only, just input vs output and error. Note that we can use the interface as the field type so the test cases can expect both the OCI and non-OCI result.